### ISSUE TITLE
error: required argument is not an integer

### ISSUE THREAD

**Issue by muhdaniyal252** [tensorflow/tensorflow#75089]
# 1. System information

- Ubuntu 22.04 (L40 GPU)
- pip package
- Tensorflow 2.13.0, tflite-support 0.4.4

# 2. Code

## Input and Output shape

 Input: \[128,32,1\]

 Output: \[1\]

## Model Architecture and Training and saving
```
import tensorflow as tf

def get_model(
        input_shape,
        output_neurons=1,
        output_activation='sigmoid',
        loss=tf.keras.losses.binary_crossentropy,
        lr=0.0001
):
    _input = tf.keras.layers.Input(shape=input_shape)
    x = tf.keras.layers.Conv2D(512,kernel_size=3,padding='valid',activation='relu')(_input)
    x = tf.keras.layers.Conv2D(256,kernel_size=3,padding='valid',activation='relu')(x)
    x = tf.keras.layers.MaxPool2D((2,2))(x)
    x = tf.keras.layers.Conv2D(128,kernel_size=3,padding='valid',activation='relu')(x)
    x = tf.keras.layers.Dropout(0.5)(x)
    x = tf.keras.layers.Conv2D(128,kernel_size=3,padding='valid',activation='relu')(x)
    x = tf.keras.layers.MaxPool2D((2,2))(x)
    x = tf.keras.layers.Conv2D(64,kernel_size=3,padding='valid',activation='relu')(x)
    x = tf.keras.layers.MaxPool2D((2,2))(x)
    x = tf.keras.layers.Flatten()(x)
    x = tf.keras.layers.Dense(1024,activation='relu')(x)
    x = tf.keras.layers.Dropout(0.3)(x)
    x = tf.keras.layers.Dense(1024,activation='relu')(x)
    x = tf.keras.layers.Dropout(0.5)(x)
    x = tf.keras.layers.Dense(1024,activation='relu')(x)
    x = tf.keras.layers.Dropout(0.7)(x)
    x = tf.keras.layers.Dense(1024,activation='relu')(x)
    x = tf.keras.layers.Dense(10,activation='relu')(x)
    outputs = tf.keras.layers.Dense(output_neurons,activation=output_activation,kernel_regularizer=tf.keras.regularizers.L2(l2=0.01))(x)
    model = tf.keras.Model(inputs=_input,outputs=outputs)

    model.compile(
        loss=loss,
        optimizer=tf.keras.optimizers.Adam(learning_rate=lr),
        metrics=['accuracy'],
    )

    return model

model = get_model(
        input_shape=input_shape,
        lr=0.001
)

reduce_lr = tf.keras.callbacks.ReduceLROnPlateau(monitor='val_accuracy',factor=0.1,patience=5,mode='max')
early_stopping = tf.keras.callbacks.EarlyStopping(monitor='val_accuracy',patience=1,mode='max',restore_best_weights=True,start_from_epoch=10)
with tf.device('/gpu'):
    history = model.fit(train,epochs=3,validation_data=val,verbose=1,callbacks=[reduce_lr,early_stopping])

model.save('model.keras')

```

## Conversion to tf lite model

```
import tensorflow as tf

model_path = 'model.keras'
lite_model_path = model_path.replace('keras','tflite')

model = tf.keras.models.load_model(model_path)

converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()

with open(lite_model_path, 'wb') as f:
    f.write(tflite_model)
```

## Adding metadata

```
actual_model = "model.keras"
model_file = "model.tflite"
export_model_file = "model_meta.tflite"

from tflite_support import metadata_schema_py_generated as _metadata_fb
from tflite_support import metadata as _metadata
from tflite_support.metadata_writers import writer_utils
import flatbuffers

model_meta = _metadata_fb.ModelMetadataT()
model_meta.name = "Binary Classification Model"
model_meta.description = "A CNN-based binary classification model for audio data."
model_meta.version = "v1"

input_meta = _metadata_fb.TensorMetadataT()
input_meta.name = "Input Tensor"
input_meta.description = (
    "Input to the model is a Mel spectrogram of audio, represented as an array of shape [1,128,32,1]."
)
input_meta.content = _metadata_fb.ContentT()
input_meta.content.contentProperties = _metadata_fb.AudioPropertiesT()
input_meta.content.contentProperties.sampleRate = 16000  # Update based on your actual sample rate
input_meta.content.contentPropertiesType = _metadata_fb.ContentProperties.AudioProperties
input_meta.shape = [1, 128, 32, 1]
input_meta.dtype = 'float32'

output_meta = _metadata_fb.TensorMetadataT()
output_meta.name = "Output Tensor"
output_meta.description = "Output is a float value between 0 and 1 representing the probability of the positive class."
output_meta.content = _metadata_fb.ContentT()
output_meta.content.contentPropertiesType = _metadata_fb.ContentProperties.FeatureProperties
output_meta.content.range = _metadata_fb.ValueRangeT()
output_meta.content.range.min = 0.0
output_meta.content.range.max = 1.0
output_meta.shape = [1]
output_meta.dtype = 'float32'

subgraph = _metadata_fb.SubGraphMetadataT()
subgraph.inputTensorMetadata = [input_meta]
subgraph.outputTensorMetadata = [output_meta]

model_meta.subgraphMetadata = [subgraph]

builder = flatbuffers.Builder(0)
meta_offset = model_meta.Pack(builder)
builder.Finish(
    meta_offset,
    _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER
)
metadata_buf = builder.Output()

populator = _metadata.MetadataPopulator.with_model_file(model_file)
populator.load_metadata_buffer(metadata_buf)
populator.load_associated_files(["labels.txt"])
populator.populate()

displayer = _metadata.MetadataDisplayer.with_model_file(export_model_file)
export_json_file = os.path.join(os.path.dirname(export_model_file), "metadata.json")
json_file = displayer.get_metadata_json()
with open(export_json_file, "w") as f:
    f.write(json_file)

```

# Description

I am developing a keyword spotting (binary classification) model that is later needed to be converted to tf lite and used in mobile device. 

For that, I built a model architecture (given above in code section). Trained is on mel spectrogram (generated using librosa). After training, I saved the model, converted to tf lite using the code mentioned above. 

Now I am at the point where i need to add metadata to the model. For that, I am refering the code given on official [documentation](https://ai.google.dev/edge/litert/models/metadata), with some necessory changes.

# Error Facing

I am facing the error at the time of adding meta data to the tflite model. When it comes to this point 
```
builder = flatbuffers.Builder(0)
meta_offset = model_meta.Pack(builder)
builder.Finish(
    meta_offset,
    _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER
)
metadata_buf = builder.Output()
```

It raises the error and exits. 

The error is 
```error: required argument is not an integer ```

I am unable to sort that error. Kindly assist me regarding that and provide me some solution for my problem.

**Comment by gaikwadrahul8** [comment]
Hi, @muhdaniyal252 

Thank you for bringing this issue to our attention and I see you've passed the `0 `value as argument while calling to `builder = flatbuffers.Builder(0)` which is by default set to `initialSize=1024` which initializes a Builder of size 1024 bytes as `initialSize` and as per documentation string **`The internal buffer is grown as needed. flatbuffers: Cannot create Builder larger than 2 gigabytes.`**

please refer this source code [link](https://github.com/google/flatbuffers/blob/8db59321d9f02cdffa30126654059c7d02f70c32/python/flatbuffers/builder.py#L92) 

Please set the initial size to a reasonable value based on your expected data size. Even if you're unsure of the exact size value like 1024 (1 KB) is often a good starting point. If your data size is unpredictable flatBuffers will automatically increase the buffer size as needed. 

Please try with `initialSize=1024` bytes like `builder = flatbuffers.Builder(1024)` and see is it resolving your issue or not ? 

If issue still persists please let us know with error log to investigate this issue further from our end. 

Thank you for your cooperation and patience.

**Comment by muhdaniyal252** [comment]
Hello @gaikwadrahul8, your response is really appreciated.

The error does not occur when I initialize the flatbuffer. It occurs after that right in the next line.

```meta_offset = model_meta.Pack(builder)```

Although I did change the initialSize to 1024, but the issue is as it is.

I am attaching the logs to backtrack the error as well.

```
error                                     Traceback (most recent call last)
Cell In[8], [line 2](vscode-notebook-cell:?execution_count=8&line=2)
      [1](vscode-notebook-cell:?execution_count=8&line=1) builder = flatbuffers.Builder(1024)
----> [2](vscode-notebook-cell:?execution_count=8&line=2) meta_offset = model_meta.Pack(builder)
      [3](vscode-notebook-cell:?execution_count=8&line=3) builder.Finish(
      [4](vscode-notebook-cell:?execution_count=8&line=4)     meta_offset,
      [5](vscode-notebook-cell:?execution_count=8&line=5)     _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER
      [6](vscode-notebook-cell:?execution_count=8&line=6) )
      [7](vscode-notebook-cell:?execution_count=8&line=7) metadata_buf = builder.Output()

File /shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:3212, in ModelMetadataT.Pack(self, builder)
   [3210](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:3210) subgraphMetadatalist = []
   [3211](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:3211) for i in range(len(self.subgraphMetadata)):
-> [3212](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:3212)     subgraphMetadatalist.append(self.subgraphMetadata[i].Pack(builder))
   [3213](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:3213) ModelMetadataStartSubgraphMetadataVector(builder, len(self.subgraphMetadata))
   [3214](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:3214) for i in reversed(range(len(self.subgraphMetadata))):

File /shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2912, in SubGraphMetadataT.Pack(self, builder)
   [2910](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2910) outputTensorMetadatalist = []
   [2911](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2911) for i in range(len(self.outputTensorMetadata)):
-> [2912](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2912)     outputTensorMetadatalist.append(self.outputTensorMetadata[i].Pack(builder))
   [2913](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2913) SubGraphMetadataStartOutputTensorMetadataVector(builder, len(self.outputTensorMetadata))
   [2914](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2914) for i in reversed(range(len(self.outputTensorMetadata))):

File /shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2325, in TensorMetadataT.Pack(self, builder)
   [2323](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2323)     dimensionNames = builder.EndVector()
   [2324](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2324) if self.content is not None:
-> [2325](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2325)     content = self.content.Pack(builder)
   [2326](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2326) if self.processUnits is not None:
   [2327](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:2327)     processUnitslist = []

File /shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:925, in ContentT.Pack(self, builder)
    [923](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:923)     contentProperties = self.contentProperties.Pack(builder)
    [924](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:924) if self.range is not None:
--> [925](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:925)     range = self.range.Pack(builder)
    [926](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:926) ContentStart(builder)
    [927](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:927) ContentAddContentPropertiesType(builder, self.contentPropertiesType)

File /shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:811, in ValueRangeT.Pack(self, builder)
    [809](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:809) ValueRangeStart(builder)
    [810](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:810) ValueRangeAddMin(builder, self.min)
--> [811](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:811) ValueRangeAddMax(builder, self.max)
    [812](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:812) valueRange = ValueRangeEnd(builder)
    [813](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:813) return valueRange

File /shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:769, in ValueRangeAddMax(builder, max)
    [768](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:768) def ValueRangeAddMax(builder, max):
--> [769](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/tensorflow_lite_support/metadata/metadata_schema_py_generated.py:769)     builder.PrependInt32Slot(1, max, 0)

File /shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:621, in Builder.PrependInt32Slot(self, *args)
--> [621](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:621) def PrependInt32Slot(self, *args): self.PrependSlot(N.Int32Flags, *args)

File /shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:602, in Builder.PrependSlot(self, flags, o, x, d)
    [600](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:600)     N.enforce_number(d, flags)
    [601](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:601) if x != d or (self.forceDefaults and d is not None):
--> [602](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:602)     self.Prepend(flags, x)
    [603](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:603)     self.Slot(o)

File /shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:594, in Builder.Prepend(self, flags, off)
    [592](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:592) def Prepend(self, flags, off):
    [593](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:593)     self.Prep(flags.bytewidth, 0)
--> [594](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:594)     self.Place(off, flags)

File /shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:762, in Builder.Place(self, x, flags)
    [760](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:760) N.enforce_number(x, flags)
    [761](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:761) self.head = self.head - flags.bytewidth
--> [762](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/builder.py:762) encode.Write(flags.packer_type, self.Bytes, self.Head(), x)

File /shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/encode.py:42, in Write(packer_type, buf, head, n)
     [40](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/encode.py:40) def Write(packer_type, buf, head, n):
     [41](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/encode.py:41)     """ Write encodes `n` at buf[head] using `packer_type`. """
---> [42](https://vscode-remote+ssh-002dremote-002bcvm.vscode-resource.vscode-cdn.net/shareddrive/working/env/lib/python3.10/site-packages/flatbuffers/encode.py:42)     packer_type.pack_into(buf, head, n)

error: required argument is not an integer
```

**Comment by gaikwadrahul8** [comment]
Hi, @muhdaniyal252

Thank you for trying, if possible could you please help us with Google colab notebook along with dataset to replicate the same behavior from our end to investigate this issue further ? Thank you.

**Comment by muhdaniyal252** [comment]
Hi, @gaikwadrahul8 

https://colab.research.google.com/drive/1daZ95UcBm2UQbRLh-d76ySDctTSd8Ydi

This is the link of the colab notebook. 

The only difference between this and my working environment is that I am using actual audios for Keyword spotting and in this case, I have created dummy data. 

The error is same here as well.

**Comment by muhdaniyal252** [comment]
Hey @gaikwadrahul8,

I were wondering if there is any update regarding the issue I am facing,

**Comment by gaikwadrahul8** [comment]
Hi, @muhdaniyal252

I apologize for the delayed response, thank you for providing the Google colab notebook and I was able to replicate the same behavior from our end so we'll have to dig more into this issue. 

Here is [gist-file](https://colab.sandbox.google.com/gist/gaikwadrahul8/6e77d8830d50d89f0ea1a309d34c8b94/test-issue-75089.ipynb) for reference

Thank you for your cooperation and patience.

**Comment by muhdaniyal252** [comment]
Hello @gaikwadrahul8,

I was wondering if there is any update regarding the issue.

**Comment by gaikwadrahul8** [comment]
Hi, @pkgoogle

Please take look into this issue. Thank you

**Comment by pkgoogle** [comment]
Hi @muhdaniyal252, tfllite_support is a little out-dated, do you absolutely need to add metadata to your model? Or do you just care about inferencing on your mobile or edge device?

**Comment by muhdaniyal252** [comment]
Hi @pkgoogle. 
I don't really care about metadata. I just need to run the model on the edge device. 
But it happens to raise error without the metadata, according to the android dev.
Do let me know if it is possible to run the model on mobile (both android and iOS) without adding metadata.

**Comment by pkgoogle** [comment]
Hi @muhdaniyal252, it seems like you already have a .tflite model? Where does it mention that an error will be raised? (it may not be incorrect but I need to understand the entire context better). You should be able to run a model on mobile w/o this step -- you may wish to look at some of our examples at LiteRT: https://github.com/google-ai-edge/litert-samples/tree/main/examples but let us know if you get stuck or an error is raised.

**Comment by muhdaniyal252** [comment]
Hi @pkgoogle, apologies for delayed response
One of the devs of android (who has done this before) told me to add metadata to the model when I provided the one without metadata. He mentioned that the error he is facing on the current model (the one without metadata) as also been previously observed and that was resolved by adding metadata to it. 
Upon investigating, I found that the tflite-support is a little back on time, which is obviously the reason of incompatibility with latest tensorflow version. 
To overcome the version compatibility conflict, I downgraded to 2.13, the latest version at the time of tflite-support last release.
That's the reason I am using the old version of Tensorflow. 

But as you said, I wouldn't need to add metadata to the model, I'll share the link with the dev and update on you very in a short while. 

Thankyou for your patience.
